net/node/node.go

```diff
@@ -105,15 +107,24 @@ func InitNode(pubKey *crypto.PubKey, nodeType int) Noder {
 	n.publicKey = pubKey
 	n.TXNPool.init()
 	n.eventQueue.init()
-
+	n.nodeDisconnectSubscriber = n.eventQueue.GetEvent("disconnect").Subscribe(events.EventNodeDisconnect, n.NodeDisconnect)
 	go n.initConnection()
 	go n.updateNodeInfo()
 
 	return n
 }
 
+func (n *node) NodeDisconnect(v interface{}) {
+	if node, ok := v.(*node); ok {
+		node.SetState(INACTIVITY)
+		conn := node.getConn()
+		conn.Close()
+	}
+}
+
 func rmNode(node *node) {
 	log.Debug(fmt.Sprintf("Remove unused/deuplicate node: 0x%0x", node.id))
+	node.local.eventQueue.GetEvent("disconnect").UnSubscribe(events.EventNodeDisconnect, node.nodeDisconnectSubscriber)
```

Comment:
the rmNode will be called for each node be destroyed, but the disconnect subscribe only be hooked for local node. Does it correctly to call the unsubscribe for each node be destroyed?